presto-docs/src/main/sphinx/connector/hive.rst

```diff
@@ -447,6 +447,83 @@ Property Name                                        Description
                                                      defaults to ``20``.
 ==================================================== ============================================================
 
+Performance Tuning Configuration Properties
+-------------------------------------------
+
+The following table describes performance tuning properties for the Hive
+connector.
+
+.. warning::
+
+   Performance tuning configuration properties are considered expert-level
+   features. Altering these properties from their default values is likely to
+   cause instability and performance degradation. We strongly suggest that
+   you use them only to address non-trivial performance issues, and that you
+   keep a backup of the original values, if you change them. 
+
+.. list-table:: Hive performance tuning configuration properties
+    :widths: 30, 50, 20
+    :header-rows: 1
+
+    * - Property name
+      - Description
+      - Default value
+    * - ``hive.domain-compaction-threshold``
+      - Sets the number of transactions to act as threshold. Once reached the
+        connector initiates compaction of the underlying files and the delta
+        files. A higher compaction threshold means reading less data from the
+        underlying data source, but a higher memory and network consumption.
+      - 100
+    * - ``hive.max-outstanding-splits``
+      - The target number of buffered splits for each table scan in a query,
+        before the scheduler tries to pause.
+      - 1000
+    * - ``hive.max-splits-per-second``
+      - Sets the maximum number of splits used per second to access underlying
+        storage. Reduce this number if your limit is routinely exceeded, based
+        on your filesystem limits. This is set to the absolute maximum value,
+        which results in Presto maximizing the parallelization of data access
+        by default. Attempting to set it higher results in Presto not being
+        able to start.
+      - Integer.MAX_VALUE
+    * - ``hive.max-initial-splits``
+      - For each query, the coordinator assigns file sections to read first
+        at the ``initial-split-size`` until the ``max-initial-splits`` is
+        reached. Then, it starts issuing reads of the ``max-split-size`` size.
+      - 200
+    * - ``hive.max-initial-split-size``
+      - Sets the initial data size for a single read section assigned to a
+        worker until ``max-initial-splits`` have been processed. You can also
+        use the catalog session property 
+        ``<catalog-name>.max_initial_split_size``.
+      - ``32 MB``
+    * - ``hive.max-split-size``
+      - Sets the largest data size for a single read section assigned to a
+        worker. You can also use the corresponding catalog session property
+        ``<catalog-name>.max_split_size``.
+      - ``64 MB``
+
+Parquet Session Properties
+--------------------------
+
+The following table describes properties supported by the Hive connector 
+to improve support for Parquet files. 
```

Comment:
The following table describes catalog session properties supported by the Hive connector to configure processing of Parquet files.